test/controllers/concerns/authentication_test.rb

```diff
@@ -3,23 +3,48 @@
 require 'test_helper'
 
 # This class tests a "dummy" controller for authentication.
-# Since testing Dummy controllers became quite complicated with
-# ActionDispatch::IntegrationTest, it is testing the Profiles controller
-# instead for the time being
 class AuthenticationTest < ActionDispatch::IntegrationTest
   setup do
-    User.current = nil
+    class AuthenticatedMockController < ActionController::API
+      include Authentication
+
+      def index
+        user = User.current
+        return false unless user
+
+        render plain: user.account_number
+      end
+
+      def raising
+        raise 'Error'
+      end
+    end
+
+    Object.const_set(:AuthenticatedMockController, AuthenticatedMockController)
+    # rubocop:disable Rails/HttpPositionalArguments
+    Rails.application.routes.draw do
+      root 'authenticated_mock#index'
+      get '/raising', to: 'authenticated_mock#raising'
+    end
```

Comment:
It really shouldn't be necessary to mess with the routes. You can instantiate a controller directly (i.e. `c = AuthenticatedMockController.new`) and then call the actions directly (i.e. `c.index`), and the before/after action callbacks should work as expected.

Reply:
I'm not sure about that, but can try to. I've seen some process methods in Rails. I've spent too much time on this mocking...

Reply:
This isn't quite what you're looking for, but I found an example in the rendering_test: https://github.com/RedHatInsights/compliance-backend/blob/master/test/controllers/concerns/rendering_test.rb#L8

Reply:
Still had to set up routes, but now via `@routes` and changed it to `ActionController::TestCase` with less magic. Unfortunately there isn't any headers passing, so those had to be added to a mocked request. Tests run much quicker with these changes.

If you still want to change it, feel free to suggest a working change. I've spent way too much time on tests compared to the actual backend change. :no_mouth:

Reply:
No worries :) I guess there isn't much benefit from continuing to iterate on these tests.